### PR TITLE
[feat] using spring cloud replace eureka in code

### PR DESCRIPTION
**Pull request by jackxu2011**
### What is the purpose of the change

there are many code using eureka for spring cloud service discovery, using config to replace the codes. And get more Extensibility

### Related issues/PRs

Related issues: #3811

### Brief change log

- remove the code for eureka, and put the corespond config file.
- using eureka config to make the service discovery in time, and remove the manual refresh eureka metadata.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Comment by casionone** [comment]
This is a big modification feature. 
In order to better understand the motivation and ideas of modification, it is necessary to create a corresponding LPIK or issue about detail info

**Comment by jackxu2011** [comment]
> This is a big modification feature. In order to better understand the motivation and ideas of modification, it is necessary to create a corresponding LPIK or issue about detail info

ok

**Comment by jacktao007** [comment]
👍

File: linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  override def getDWCServiceInstance(
+      serviceInstance: SpringCloudServiceInstance
+  ): ServiceInstance = {
+    val applicationName = serviceInstance.getServiceId
+    val instanceId = serviceInstance.getInstanceId
+    logger.info("service name: {}, instance id: {}", applicationName: Any, instanceId: Any)
+    ServiceInstance(applicationName, getInstance(serviceInstance))
+  }
+
+  private[rpc] def getInstance(serviceInstance: SpringCloudServiceInstance): String =
```
Comment (peacewong) [inline linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala:42]: can remove

**Review by peacewong — APPROVED** [review]

**Review by peacewong — APPROVED** [review]
LGTM.

File: linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala [hunk 2]
```diff
@@ -15,25 +15,26 @@
...
+import org.apache.linkis.rpc.interceptor.AbstractRPCServerLoader
 
-import scala.collection.JavaConverters._
+import scala.concurrent.duration.Duration
 
-trait DiscoveryMsgGenerator {
+class SpringRPCServerLoader extends AbstractRPCServerLoader {
 
-  def generate(engineConnManagerEnv: EngineConnManagerEnv): java.util.Map[String, String]
+  override def refreshAllServers(): Unit = {}
```
Comment (peacewong) [inline linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala:28]: The implementation of eureka should be kept

Comment (wForget) [inline linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala:28]: We still need to refresh the discovery client like [EurekaClientRefreshUtils.refreshEurekaClient()](https://github.com/apache/incubator-linkis/blob/ac050be75992b114d806814614b8980adfe5e18b/linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/eureka/EurekaClientRefreshUtils.scala#L41) method when not getting a server.

Comment (wForget) [reply linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala:28]: Maybe we can add a `DiscoveryClientRefreshUtils` class to implement different refresh methods according to the type of discoveryClient.

Comment (jackxu2011) [reply linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala:28]: I use some eureka config to replace the refresh method.  add the refresh is sometime useless. the eureka's cache can be controlled by eureka's config.   On the other hand,  the eureka's refresh method is used for test only.

Comment (wForget) [reply linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala:28]: We need to trigger a refresh when we don't get a server, not an interval refresh.

File: linkis-dist/package/conf/application-engineconn.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    registry-fetch-interval-seconds: 8
+    initial-instance-info-replication-interval-seconds: 10
+management:
+  endpoints:
+    web:
+      exposure:
+        include: refresh,info,health,metrics
+
+logging:
+  config: classpath:log4j2.xml
```
Comment (wForget) [inline linkis-dist/package/conf/application-engineconn.yml:43]: nit: newline

File: linkis-spring-cloud-services/linkis-service-gateway/plugins/linkis-gateway-datasource-ruler/src/main/scala/org/apache/linkis/gateway/ruler/datasource/DatasourceGatewayRouterRuler.scala [hunk 4]
```diff
@@ -86,4 +93,4 @@ object DatasourceGatewayRouterRuler {
 
   val DATASOURCE_NAME_KEY = "wds.linkis.datasource"
 
-}
\ No newline at end of file
+}
```
Comment (wForget) [inline linkis-spring-cloud-services/linkis-service-gateway/plugins/linkis-gateway-datasource-ruler/src/main/scala/org/apache/linkis/gateway/ruler/datasource/DatasourceGatewayRouterRuler.scala:96]: nit: newline

File: linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala (hunk 2 above)
Comment (jackxu2011) [reply linkis-commons/linkis-rpc/src/main/scala/org/apache/linkis/rpc/sender/spring/SpringRPCServerLoader.scala:28]: 原来需要通过手动刷新缓存是因为eureka的缓存刷新时间间隔是30s，rpc获取ec的最长等待时间也是30s会还在拿不到ec的情况，把eureka的缓存刷新时间改为8s后，理论上是不会出现这个情况了。

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**